Library/Homebrew/software_spec.rb

```diff
@@ -402,22 +404,37 @@ def compatible_locations?
   # Does the {Bottle} this {BottleSpecification} belongs to need to be relocated?
   sig { returns(T::Boolean) }
   def skip_relocation?
-    cellar == :any_skip_relocation
+    legacy_cellar_or_cellar_for_tag == :any_skip_relocation
   end
 
   def tag?(tag)
     checksum_for(tag) ? true : false
   end
 
-  # Checksum methods in the DSL's bottle block optionally take
+  # BottleChecksum methods in the DSL's bottle block take
   # a Hash, which indicates the platform the checksum applies on.
-  Checksum::TYPES.each do |cksum|
-    define_method(cksum) do |val|
-      digest, tag = val.shift
-      collector[tag] = Checksum.new(cksum, digest)
+  # Example args:
+  # {"22948764d8f8d7be4870ff92dae64d986eb63a9150b219c20fff87d1a6aa93d6"=>
+  # {"tag"=>:mojave, "cellar"=>:any_skip_relocation}}
+  Checksum::TYPES.each do |cksum_type|
+    define_method(cksum_type) do |*args|
+      bottle_hash = args.first
+      tag = bottle_hash.values.first
+      # Legacy DSL: There is only a single cellar line for all the tags
+      # New DSL: get the cellar for the specific tag
+      cellar_for_tag = cellar.presence || bottle_hash.values.second
```

Comment:
```suggestion
      # Legacy DSL: There is only a single cellar line for all the tags
      # New DSL: get the cellar for the specific tag
      cellar_for_tag = cellar.presence || bottle_hash.values.second
```

Providing an example for the DSL here (and in other cases) will help make it more obvious why it's `.second` (which I think may warrant a refactor)

Reply:
Shouldn't this be `bottle_hash["cellar"]`? You can't rely on the `values` of a hash being ordered how you want.

Reply:
Yes, it is  bottle_hash["cellar"]. I change the code to try to make it clearer.